youtube_dl/extractor/jwplatform.py

```diff
@@ -50,7 +50,11 @@ def _parse_jwplayer_data(self, jwplayer_data, video_id=None, require_title=True,
             # JWPlayer backward compatibility: flattened sources
             # https://github.com/jwplayer/jwplayer/blob/v7.4.3/src/js/playlist/item.js#L29-L35
             if 'sources' not in video_data:
-                video_data['sources'] = [video_data]
+                if isinstance(video_data, list):
```

Comment:
This is incorrect and may break other websites. According to [1], an array _is_ a playlist. Codes for handling incorrect JWPlayer usage in IQM2 should be in ```iqm2.py```.

[1] https://support.jwplayer.com/customer/portal/articles/1481963-building-managing-playlists#fndtn-code